lib/puppetserver/ca/config/puppet.rb

```diff
@@ -152,12 +152,20 @@ def resolve_settings(overrides = {})
             settings[setting_name] = setting_value
           end
 
+          # If user provides subject-alt-names, we need to add the certname in addition
+          # to these.
+          if overrides[:dns_alt_names]
+            overrides[:dns_alt_names] = overrides[:dns_alt_names] + ',' + '$certname'
+          end
           # rename dns_alt_names to subject_alt_names now that we support IP alt names
-          settings[:subject_alt_names] = overrides.fetch(:dns_alt_names, "puppet,$certname")
+          settings[:subject_alt_names] = overrides.fetch(:dns_alt_names, "")
+          # If the cert is a master cert, it requires different defaults:
+          settings[:ca_subject_alt_names] = overrides.fetch(:dns_alt_names, "puppet,$certname")
 
           # Some special cases where we need to manipulate config settings:
           settings[:ca_ttl] = munge_ttl_setting(settings[:ca_ttl])
           settings[:certificate_revocation] = parse_crl_usage(settings[:certificate_revocation])
+          settings[:ca_subject_alt_names] = munge_alt_names(settings[:ca_subject_alt_names])
```

Comment:
So, my options were to either provide overrides from the actions, or to do the work here and create a separate SANs for master certs.  I called it ```ca_subject_alt_names```, but I could call it ```master_subject_alt_names``` too.    Ideas to make this better are welcomed.